// @syncfusion/ej2-ng-buttons/src/switch/switch.component.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Component, ElementRef, ViewContainerRef, Renderer2, ChangeDetectionStrategy, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { ComponentBase, applyMixins, FormBase, setValue } from '@syncfusion/ej2-ng-base';
import { Switch } from '@syncfusion/ej2-buttons';
export var inputs = ['checked', 'cssClass', 'disabled', 'enablePersistence', 'enableRtl', 'locale', 'name', 'offLabel', 'onLabel', 'value'];
export var outputs = ['focus', 'blur', 'change', 'checkedChange'];
export var twoWays = ['checked'];
/**
 * Represents the Angular Switch Component.
 * ```html
 * <ejs-switch></ejs-switch>
 * ```
 */
var SwitchComponent = /** @class */ (function (_super) {
    __extends(SwitchComponent, _super);
    function SwitchComponent(ngEle, srenderer, viewContainerRef) {
        var _this = _super.call(this) || this;
        _this.ngEle = ngEle;
        _this.srenderer = srenderer;
        _this.viewContainerRef = viewContainerRef;
        _this.element = _this.ngEle.nativeElement;
        _this.injectedModules = _this.injectedModules || [];
        for (var i = 3; i < arguments.length; i++) {
            var injectedObj = arguments[i];
            if (injectedObj) {
                _this.injectedModules.push(injectedObj);
            }
        }
        _this.registerEvents(outputs);
        _this.addTwoWay.call(_this, twoWays);
        setValue('currentInstance', _this, _this.viewContainerRef);
        return _this;
    }
    SwitchComponent.prototype.registerOnChange = function (registerFunction) {
    };
    SwitchComponent.prototype.registerOnTouched = function (registerFunction) {
    };
    SwitchComponent.prototype.writeValue = function (value) {
    };
    SwitchComponent.prototype.setDisabledState = function (disabled) {
    };
    SwitchComponent.prototype.ngOnInit = function () {
    };
    SwitchComponent.prototype.ngAfterViewInit = function () {
    };
    SwitchComponent.prototype.ngOnDestroy = function () {
    };
    SwitchComponent.prototype.ngAfterContentChecked = function () {
    };
    SwitchComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ejs-switch',
                    inputs: inputs,
                    outputs: outputs,
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(function () { return SwitchComponent; }),
                            multi: true
                        }
                    ],
                    queries: {}
                },] },
    ];
    /** @nocollapse */
    SwitchComponent.ctorParameters = function () { return [
        { type: ElementRef, },
        { type: Renderer2, },
        { type: ViewContainerRef, },
    ]; };
    return SwitchComponent;
}(Switch));
export { SwitchComponent };
applyMixins(SwitchComponent, [ComponentBase, FormBase]);
